src/adapters/marketplace/v00.js

```diff
@@ -100,27 +100,40 @@ class V00_MarkeplaceAdapter {
   }
 
   async makeOffer(listingId, ipfsBytes, data, confirmationCallback) {
-    const {
-      finalizes,
-      affiliate,
-      commission,
-      price,
-      arbitrator,
-      currency
-    } = data
+    const { affiliate, arbitrator, commission = {}, finalizes, totalPrice = {}, unitsPurchased } = data
+    // For V1, we only support quantity of 1.
+    if (unitsPurchased != 1)
+      throw new Error(
+        `Attempted to purchase ${unitsPurchased} - only 1 allowed.`
+      )
+
+    // Convert price to correct units for blockchain
+    let currency, price
+    if (totalPrice.currency === 'ETH') {
+      currency = 'ETH'
+      price = this.contractService.web3.utils.toWei(totalPrice.amount, 'ether')
+    } else {
+      // handle ERC20
+      currency = this.contractService.currencies[totalPrice.currency]
+      // TODO consider using ERCStandardDetailed.decimals() (for tokens that support this) so that we don't have to track decimals ourselves
+      // https://github.com/OpenZeppelin/openzeppelin-solidity/blob/6c4c8989b399510a66d8b98ad75a0979482436d2/contracts/token/ERC20/ERC20Detailed.sol
+      const currencyDecimals = currency && currency.decimals
+      price = String(Number(totalPrice.amount) * 10**currencyDecimals)
```

Comment:
Doesn't this have the risk to overflow if we use regular JS Number vs BigNumber ?
Won't cause issue in V1 since this code path won't get exercised as we only support listing prices in ETH. But might break in future ?

Reply:
Good call @franckc 👍 I just copied that over but will use the `toWei` like I did in the listing logic.

Reply:
@franckc ah ok, I think you're right. I'll fix this. Thanks.